src/mono/netcore/sample/Android/AndroidSampleApp.csproj

```diff
@@ -5,25 +5,28 @@
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <TargetArchitecture Condition="'$(TargetArchitecture)'==''">x64</TargetArchitecture>
     <TargetOS>Android</TargetOS>
+    <EnableTargetingPackDownload>false</EnableTargetingPackDownload>
+    <RuntimeIdentifier>android-$(TargetArchitecture)</RuntimeIdentifier>
+    <PublishTrimmed>true</PublishTrimmed>
+    <_TrimmerDefaultAction>link</_TrimmerDefaultAction>
+    <Optimized Condition="'$(Configuration)' == 'Release'">True</Optimized>
     <MicrosoftNetCoreAppRuntimePackDir>$(ArtifactsBinDir)microsoft.netcore.app.runtime.android-$(TargetArchitecture)\$(Configuration)\runtimes\android-$(TargetArchitecture)\</MicrosoftNetCoreAppRuntimePackDir>
     <NoWarn>$(NoWarn),CA1050</NoWarn>
   </PropertyGroup>
 
-  <Target Name="RebuildAndroidAppBuilder">
+  <!-- Redirect 'dotnet publish' to in-tree runtime pack -->
+  <Target Name="TrickRuntimePackLocation" AfterTargets="ProcessFrameworkReferences">
     <ItemGroup>
-      <AndroidAppBuilderProject Include="$(RepoTasksDir)mobile.tasks\AndroidAppBuilder\AndroidAppBuilder.csproj" />
+      <RuntimePack>
+        <PackageDirectory>$(ArtifactsBinDir)microsoft.netcore.app.runtime.android-$(TargetArchitecture)\$(Configuration)</PackageDirectory>
+      </RuntimePack>
     </ItemGroup>
-    <MSBuild Projects="@(AndroidAppBuilderProject)"
-             Properties="Configuration=$(Configuration);MSBuildRestoreSessionId=$([System.Guid]::NewGuid())"
-             Targets="Restore"/>
-    <MSBuild Projects="@(AndroidAppBuilderProject)"
-             Properties="Configuration=$(Configuration)"
-             Targets="Build;Publish"/>
+    <Message Text="Packaged ID: %(RuntimePack.PackageDirectory)" Importance="high" />
   </Target>
 
   <UsingTask TaskName="AndroidAppBuilderTask" AssemblyFile="$(AndroidAppBuilderTasksAssemblyPath)"/>
 
-  <Target Name="BuildApp" DependsOnTargets="RebuildAndroidAppBuilder;Build">
+  <Target Name="BuildApp" AfterTargets="CopyFilesToPublishDirectory">
     <PropertyGroup>
       <AndroidAbi Condition="'$(Platform)'=='arm64'">arm64-v8a</AndroidAbi>
```

Comment:
this shouldn't use `Platform`, but `RuntimeIdentifier` instead (GitHub won't allow me to comment on the whole block:

```diff
-      <AndroidAbi Condition="'$(Platform)'=='arm64'">arm64-v8a</AndroidAbi>
-      <AndroidAbi Condition="'$(Platform)'=='arm'">armeabi-v7a</AndroidAbi>
-      <AndroidAbi Condition="'$(Platform)'=='x64'">x86_64</AndroidAbi>
-      <AndroidAbi Condition="'$(AndroidAbi)'==''">$(Platform)</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-arm64'">arm64-v8a</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-arm'">armeabi-v7a</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-x86'">x86</AndroidAbi>
+      <AndroidAbi Condition="'$(RuntimeIdentifier)'=='android-x64' or '$(AndroidAbi)'==''">x86_64</AndroidAbi>
```

Or probably an even better fix would be to just pass `RuntimeIdentifier` into `AndroidAppBuilderTask` and do the conversion internally there.